From 87dcdcc12434d2ed251efc182f2a24a244e74e4d Mon Sep 17 00:00:00 2001
From: Pauli <pauli@openssl.org>
Date: Thu, 14 Jul 2022 15:22:30 +1000
Subject: [PATCH 046/209] Coverity 1506566: unchecked return value

There isn't much else that can be done here unfortunately.

Reviewed-by: Tomas Mraz <tomas@openssl.org>
Reviewed-by: Dmitry Belyavskiy <beldmit@gmail.com>
Reviewed-by: Hugo Landau <hlandau@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/18799)

(cherry picked from commit 358103b4a651ab3f392f088d86cd30469dccce2e)
---
 crypto/bio/bss_dgram.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/crypto/bio/bss_dgram.c b/crypto/bio/bss_dgram.c
index aa82361306..8ca1cf64ed 100644
--- a/crypto/bio/bss_dgram.c
+++ b/crypto/bio/bss_dgram.c
@@ -1914,7 +1914,8 @@ static void get_current_time(struct timeval *t)
     t->tv_sec = (long)(now_ul / 10000000);
     t->tv_usec = ((int)(now_ul % 10000000)) / 10;
 # else
-    gettimeofday(t, NULL);
+    if (gettimeofday(t, NULL) < 0)
+        perror("gettimeofday");
 # endif
 }
 
-- 
2.34.1

